Yoast/wordpress-seo
# Add assessment for product identifiers

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Implements the Product identifier/Barcode assessment and temporarily disables it until the necessary data from Woo and Shopify become available.
* [yoastseo] Implements the Product identifier/Barcode assessment and temporarily disables it until the necessary data from Woo and Shopify become available.
* [shopify-seo]  Implements the Product identifier/Barcode assessment and temporarily disables it until the necessary data from Woo and Shopify become available.

## Relevant technical choices:

* This is the first PR needed to introduce the Product Identifier/Barcode assessment in Woo and Shopify. The assessment is implemented in this PR but it's not working yet, as we don't yet access the data from the Woo and Shopify repositories. 
This is why the applicability condition of the assessments is now set to `false`; it will be changed once the necessary data become available.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### Test that the assessment with Woo settings
* Change the applicability condition in `ProductIdentifierAssessment.js` to `true` (line 71)
* Build and activate Yoast SEO Free 
* Require the current branch in the Woo `trunk`: `composer require yoast/wordpress-seo:dev-PC-15-add-assessment-for-prodcut-identifiers@dev` and Activate Yoast SEO Woo
* Open/create a product 
* Confirm that the product identifier assessment appears in the SEO metabox
* Go the Inspect console
* Try and add some mocked data by typing the following script:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
    hasGlobalIdentifier: true,
    hasVariants: false,
  } );
  return data;
} );
```
* Add some text in the editor to make the assessment update
* Confirm that the assessment returns a green bullet and the feedback: `Product identifier: Good job!`
* Change the data to the following:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
  hasGlobalIdentifier: false,
hasVariants: true,
doAllVariantsHaveIdentifier: true,
  } );
  return data;
} );
```
* Confirm that you get a green bullet and the feedback: `Product identifier: Good job!`


#### Test the assessment with Shopify settings
* Change the parameters values for the Product Identifier assessment in `Assessor.js` from this:
```
new ProductIdentifiersAssessment( {
			urlTitle: createAnchorOpeningTag( options.productIdentifierUrlTitle ),
			urlCallToAction: createAnchorOpeningTag( options.productIdentifierCTAUrl ),
			assessVariants: options.assessVariants,
			productIdentifierOrBarcode: options.productIdentifierOrBarcode,
		} ),
```
to this:
```
new ProductIdentifiersAssessment( {
urlTitle: createAnchorOpeningTag( “https://yoa.st/4ly” ),
			urlCallToAction: createAnchorOpeningTag( “https://yoa.st/4lz” ),
			assessVariants: false,
			productIdentifierOrBarcode: { lowercase: “barcode” },
} ),
```
* Build and activate Yoast SEO Free 
* Open/create a new product
* Go the Inspect console
* Try and add some mocked data by typing the following script:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
    hasGlobalIdentifier: true,
    hasVariants: false,
  } );
  return data;
} );
```
* Add some text in the editor to make the assessment update
* Confirm that the assessment returns a green bullet and the feedback: `Barcode: Good job!`
* * Change the data to the following:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
  hasGlobalIdentifier: false,
hasVariants: true,
  } );
  return data;
} );
```
* Confirm that the assessment does not appear anymore.

### Test instructions for QA when the code is in the RC



QA can test this PR by following these steps:

* Install and activate Yoast SEO Free and Woo
* Open/create a new product
* Confirm that the Product identifier/Barcode assessment is not showing in the SEO assessments section
* Install and activate Yoast SEO for Shopify app
* Open/create a new product
* Confirm that the Product identifier/Barcode assessment is not showing in the SEO assessments section

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/PC-134
